PhysicsTools/Heppy/python/analyzers/objects/PhotonAnalyzer.py

```diff
@@ -135,7 +152,48 @@ def idWP(gamma,X):
         self.counters.counter('events').inc('all events')
         if foundPhoton: self.counters.counter('events').inc('has >=1 gamma at preselection')
         if len(event.selectedPhotons): self.counters.counter('events').inc('has >=1 selected gamma')
-       
+
+    def checkGenPhoton(self, event):
+        def isHardParton(particle):
+            return abs(particle.pdgId()) in [1,2,3,4,5,6,21] and particle.status() == 23
+
+        def isPromptPhoton(photon):
+            return photon.status() == 1 and abs(photon.mother(0).pdgId()) <= 22 and abs(photon.mother(0).pdgId()) not in [11,13,15]
```

Comment:
Here, FSR photons from prompt leptons will not be flagged as prompt, while the `GenParticle` status flags like  `isPromptFinalState` would call them prompt (and they would be isolated, so calling them prompt is probably right).
Maybe can you call it `isPromptPhotonNoFSR` or something similar?
Or, what are you trying to achieve?
I think it would be good to have a more solid definition of what we call mc truth photons in heppy, and if for whatever reason we need a more ad-hoc definition because it's used in susy then it can be put somewhere else (or labelled somehow)
Tagging @bendavid which just told me that not all FSR photons will have a lepton as a mother if they're added by Madgraph and not Pythia